src/CommunityToolkit.Maui/ImageSources/GravatarImageSource.shared.cs

```diff
@@ -0,0 +1,252 @@
+	public TimeSpan CacheValidity
+	{
+		get => (TimeSpan)GetValue(CacheValidityProperty);
+		set => SetValue(CacheValidityProperty, value);
+	}
+
+	/// <summary>Gets or sets the size of the gravatar image.</summary>
+	public int Size
+	{
+		get => (int)GetValue(SizeProperty);
+		set => SetValue(SizeProperty, value);
+	}
+
+	/// <summary>Gets or sets a Boolean value that indicates whether caching is enabled on this <see cref="GravatarImageSource"/> object.</summary>
+	public bool CachingEnabled
+	{
+		get => (bool)GetValue(CachingEnabledProperty);
+		set => SetValue(CachingEnabledProperty, value);
+	}
+
+	/// <summary>Gets or sets the email address.</summary>
+	public string? Email
+	{
+		get => (string)GetValue(EmailProperty);
+		set => SetValue(EmailProperty, value);
+	}
+
+	/// <summary>Gets or sets the default image.</summary>
+	public DefaultImage Image
+	{
+		get => (DefaultImage)GetValue(ImageProperty);
+		set => SetValue(ImageProperty, value);
+	}
+
+	/// <summary>Gets a value indicating whether the control email is empty.</summary>
+	public override bool IsEmpty => string.IsNullOrEmpty(Email);
+
+	/// <summary>Gets or sets the URI for the image to get.</summary>
+	[System.ComponentModel.TypeConverter(typeof(UriTypeConverter))]
+	public Uri Uri { get; private set; }
+
+	/// <summary>Gets or sets the control size property.</summary>
+	int GravatarSize
+	{
+		get => gravatarSize;
+		set => gravatarSize = Math.Clamp(value, 1, 2048);
+	}
+
+	/// <summary>Returns the Uri as a string.</summary>
+	/// <returns>String of the URI.</returns>
+	public override string ToString()
+	{
+		return $"Uri: {Uri}, Email:{Email}, Size:{GravatarSize}, Image: {DefaultGravatarName(Image)}, CacheValidity:{CacheValidity}, CachingEnabled:{CachingEnabled}";
+	}
+
+	/// <summary>On parent set.</summary>
+	protected override void OnParentSet()
+	{
+		object element = Parent;
+		base.OnParentSet();
+		if (element is not VisualElement parentElement || parentElement is null)
+		{
+			GravatarSize = defaultSize;
+			return;
+		}
+
+		// Only bind if the consumer hasn't opted for a fixed Size.
+		if (Size == 0)
+		{
+			SetBinding(SizeProperty, new Binding(nameof(VisualElement.Width), source: parentElement));
+			SetBinding(SizeProperty, new Binding(nameof(VisualElement.Height), source: parentElement));
+		}
+	}
+
+	/// <summary>On property changed.</summary>
+	/// <param name="propertyName">Property name.</param>
+	protected override void OnPropertyChanged(string propertyName)
+	{
+		if (propertyName is not null && (propertyName == CacheValidityProperty.PropertyName || propertyName == CachingEnabledProperty.PropertyName || propertyName == EmailProperty?.PropertyName || propertyName == ImageProperty.PropertyName))
+		{
+			if (propertyName == CacheValidityProperty.PropertyName || propertyName == CachingEnabledProperty.PropertyName)
+			{
+				OnUriChanged();
+			}
+
+			return;
+		}
+
+		base.OnPropertyChanged(propertyName);
+	}
```

Comment:
We may be able to remove this if the `return` doesn't protect us from anything in favour of using the `BindableProperty` `propertyChanged` method.

Reply:
Although to double check, do we need to be calling `OnUriChanged` when the cache properties change?

Reply:
don't need to check for null either

Reply:
This method has been removed as the code has changed and it's no-longer needed.